### PR TITLE
add pytest migration guide back

### PR DESCRIPTION
**Pull request by mashehu**
the deploy preview will not render the docs update, if you want to see it rendered in all its glory, run `npm run dev -w sites/docs` locally

### CODE REVIEW

**Review by sateeshperi — APPROVED** [review]

File: sites/docs/src/content/docs/tutorials/nf-core_components/migrate_to_nf-test.mdx [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+-   If migrating an existing module, get the inputs from current pytest files `tests/modules/nf-core/module/main.nf` and provide as positional inputs `input[0]` in nf-test file
+
+```groovy title="main.nf.test"
+input[0] = [
+            [id:"ref"],
+            file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
+           ]
+```
+
+-   Next, in the `then` block we can write our assertions that are used to verify the test. A test can have multiple assertions but, we recommend enclosing all assertions in a `assertAll()` block as shown below:
```
Comment (jfy133) [inline sites/docs/src/content/docs/tutorials/nf-core_components/migrate_to_nf-test.mdx:62]: COuldn't we link back to the main nf-test docs tutorials etc for the remainder?

**Review by jfy133 — COMMENTED** [review]
I guess it's fine, but I find it's maybe confusing for many people (unnecessary technical info), and there is a lot of redundant info with https://nf-co.re/docs/tutorials/tests_and_test_data/nf-test_writing_tests

I'm not going to block this for now as I don't have time to re-write it, but I think this can be as simple as:

- Run this command
    - Don't worry that it says 'create', you're not duplicating your module
- here are the changes you expect to see
- Look at https://nf-co.re/docs/tutorials/tests_and_test_data/nf-test_writing_tests for how to fill in the files

File: sites/docs/src/content/docs/tutorials/nf-core_components/migrate_to_nf-test.mdx [hunk 2]
```diff
@@ -0,0 +1,215 @@
+---
+title: Migrating from pytest to nf-test
+shortTitle: Migrating to nf-test
+subtitle: Quick reference of the steps needed to add a new module to a pipeline.
```
[the remark this answers is not in the document]
Comment (mashehu) [reply sites/docs/src/content/docs/tutorials/nf-core_components/migrate_to_nf-test.mdx:4]: oops

**Comment by mashehu** [comment]
good points will rework it a bit to make it more concise

File: sites/docs/src/content/docs/tutorials/nf-core_components/migrate_to_nf-test.mdx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+-   Use nf-core/tools to migrate the module/subworkflow with `--migrate-pytest`.
+
+    <Tabs names={["modules","subworkflows"]} icons={['<i class="far fa-xs fa-code-commit me-2"></i>','<i class="far fa-code-fork fa-rotate-90 me-2"></i>']} client:idle>
+
+        <TabItem name={"modules"} client:idle>
+            ```bash
+            nf-core modules create <tool>/<subtool> --migrate-pytest
+            ```
+            :::info{title="Technical details" collapse}
+            This command will: - rename the current module directory to `<module>_old` to avoid conflicts with the new module, - create a new module named `<module>`, based on the nf-test template. - copy the `main.nf`, `meta.yml` and `environment.yml` files over to preserve the original module code. - (optional) If your module has a `nextflow.config` file to run (e.g. for `ext.args` specification), the command will also copy it to the module's `tests/` directory and the path will be added to the `main.nf.test` file.
```
Comment (mirpedrol) [inline sites/docs/src/content/docs/tutorials/nf-core_components/migrate_to_nf-test.mdx:30]: the line breaks were lost :)

File: sites/docs/src/content/docs/tutorials/nf-core_components/migrate_to_nf-test.mdx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+                -   rename the current subworkflow directory to `<subworkflow>_old` to avoid conflicts with the new subworkflow,
+                -   create a new subworkflow named `<subworkflow>` based on the nf-test template.
+                -   copy the `main.nf` and `meta.yml` files over to preserve the original subworkflow code.
+            :::
+        </TabItem>
+
+    </Tabs>
+
+-   You will then be asked if you want to delete the old files or keep them. No worries, we will print the content of the old pytests in the terminal so you copy the information to the new nf-test files.
```
Comment (mirpedrol) [inline sites/docs/src/content/docs/tutorials/nf-core_components/migrate_to_nf-test.mdx:53]:
```suggestion
-   You will then be asked if you want to delete the old files or keep them. No worries, we will print the content of the old pytests in the terminal so you can copy the information to the new nf-test files.
```

**Review by mirpedrol — APPROVED** [review]

**Review by jfy133 — APPROVED** [review]
Much better :+1: 

Only thing I would argue (But not a blocker) is It would make more sense in teh test data-tests section rather than components

image

**Pull request merged**